pkg/network/vlan/vlan.go

```diff
@@ -125,7 +125,15 @@ func (d *VlanDriver) Init() error {
 		d.vlanParentIndex = device.Attrs().ParentIndex
 		//glog.Infof("root device %s is a vlan device, parent index %d", d.Device, d.vlanParentIndex)
 	}
-	if d.MacVlanMode() || d.IPVlanMode() {
+	if d.IPVlanMode() {
```

Comment:
Thanks for fixing this. Can you help replace the if statement with 

```
if d.IPVlanMode() {
...
} else if d.MacVlanMode() {
...
} else if d.PureMode() {
...
}

```

Reply:
yeah, "else if" is more beautiful